test/test_fully_connected_layer.h

```diff
@@ -44,8 +44,8 @@ TEST(fully_connected, forward) {
   // clang-format on
   vec_t expected = {24, 42};
 
-  l.ith_parameter(0).set_data(Tensor<float_t>(weights));
-  l.ith_parameter(1).set_data(Tensor<float_t>(bias));
+  l.weights_at(0).set_data(Tensor<float_t>(weights));
```

Comment:
I wouldn't expose `weights_at(...)` to fully connected since it only has data at 0 position, it might lead to bugs. Expose a single method to get weights and bias:
```cpp
tensor fully_connected_layer::weights() const {
  return weights_at(0);
}
```
same for bias
```cpp
tensor fully_connected_layer::bias() const {
  return biases_at(0);
}
```